src/packages/@ncigdc/modern_components/GeneExpression/inchlib/index.js

```diff
@@ -2530,34 +2535,51 @@ import ageDisplay, { getLowerAgeYears } from '@ncigdc/utils/ageDisplay';
       },
     ];
 
-    const download_ul = $(`<ul class="inchlib-download"></ul>`);
+    const clickOutOverlay = self._draw_overlay(true);
+    const hideClickOutOverlay = () => {
+      clickOutOverlay.remove();
+      $('.inchlib-download').remove();
+    }
+
+    clickOutOverlay.click(hideClickOutOverlay);
+
+    const visualDownloadFeedback = (button, handler) => {
+      const loadingOverlay = self._draw_overlay();
+      const format = button.html();
+      button.html(`Processing ${format}...`).css('background-color', '#CCC');
+      self.$element.prepend(loading_div);
+
+      setTimeout(() => {
```

Comment:
we delay the download handler itself for a split second, so the loading overlay can come up first.